src/test/java/org/jenkinsci/plugins/ewm/casc/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,102 @@
+package org.jenkinsci.plugins.ewm.casc;
+
+import hudson.ExtensionList;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.ewm.definitions.Disk;
+import org.jenkinsci.plugins.ewm.definitions.DiskPool;
+import org.jenkinsci.plugins.ewm.steps.ExwsAllocateStep;
+import org.jenkinsci.plugins.ewm.steps.ExwsStep;
+import org.jenkinsci.plugins.ewm.definitions.Template;
+import org.jenkinsci.plugins.ewm.nodes.ExternalWorkspaceProperty;
+import org.jenkinsci.plugins.ewm.nodes.NodeDisk;
+import org.jenkinsci.plugins.ewm.nodes.NodeDiskPool;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import org.jvnet.hudson.test.JenkinsRule;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.snakeyaml.Yaml;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test for Configuration As Code Compatibility.
+ *
+ * @author Martin d'Anjou
+ */
+public class ConfigAsCodeTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void shouldSupportConfigurationAsCode() throws Exception {
+        URL resource = ConfigAsCodeTest.class.getResource("configuration-as-code.yaml");
```

Comment:
if you add the tests dependency you can simplify this code:
```
    <dependency>
      <groupId>io.jenkins</groupId>
      <artifactId>configuration-as-code</artifactId>
      <version>1.8</version>
      <classifier>tests</classifier>
      <scope>test</scope>
    </dependency>
```

then you annotate the test method with:
```
@ConfiguredWithCode("configuration-as-code.yaml")
```

and change the rule to be:
```
@Rule
public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
```

All the configuration as code setup bit will be done for you automatically

Reply:
I agree. First I want to debug what's wrong with the Descriptor. I prefer to work on one problem at a time. See the commit 9d342da

Reply:
I'd say, we should not refactor the discriptor. According to my obeservation, only those descriptor with a ```configure``` method will need fields in the descriptor, and all those descriptors are global configurations. Node properties is a little bit special, but I take it as a not-top-level describable, as there were no ```configure()``` method defined previously. I also guess there would be a describable serve as the top-level who appear in the web UI.

Reply:
@martinda Hi Martin, our previous direction was correct. I all push a new version tomorrow  : )

Reply:
Thanks @imaffe looking forward to it.